internal/services/mssql/mssql_virtual_machine_resource.go

```diff
@@ -285,7 +285,12 @@ func resourceMsSqlVirtualMachine() *pluginsdk.Resource {
 						},
 						"data_settings":    helper.StorageSettingSchema(),
 						"log_settings":     helper.StorageSettingSchema(),
-						"temp_db_settings": helper.StorageSettingSchema(),
+						"temp_db_settings": helper.SQLTempDBStorageSettingSchema(),
+						"sql_system_db_on_datadisk": {
```

Comment:
we can remove sql here, and we typically put enabled on booleans
```suggestion
						"system_db_on_datadisk_enabled": {
```